lib/cli.js

```diff
@@ -17,86 +17,82 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = path.resolve(gitDirectory, 'logs/incorrect-commit-msgs');
+  }
+} catch (err) {}
+
+var bufferToString = function (buffer) {
+  var hasToString = buffer && typeof buffer.toString === 'function';
+
+  return hasToString && buffer.toString();
+}
+
+var getCommit = function (msgFileOrText) {
+  var commit = getCommitFromFile(msgFileOrText);
+
+  if (commit === null && !msgFileOrText) {
```

Comment:
Ok, took a step back to rethink about it.
- we actually have 2 cases : `msgFileOrText` is provided as an argument to the cli or not
- if not, we want to look at `COMMIT_EDITMSG`
- the function must return an object containing `message` key, whatever the case

To answer your question: the check is needed because if `msgFileOrText` is provided, we do not want to default on `COMMIT_EDITMSG`.

I refactored it to make all this clearer:
```js
  if (msgFileOrText !== undefined) {
    return getCommitFromFile(msgFileOrText) || { message: msgFileOrText };
  }

  return getCommitFromFile('COMMIT_EDITMSG') || { message: null };
```
Now for the pleasure we can shorten it a bit: it's not important here if `message` is `null` or `undefined`, we did it both in our codes. So the fallbacks can be merged:
```js
  return getCommitFromFile(msgFileOrText || 'COMMIT_EDITMSG')
    || { message: msgFileOrText };
```

All these codes have the exact same result, even [those](https://github.com/conventional-changelog/validate-commit-msg/pull/109#discussion_r130929896).

So let's keep the first one in this message, it's maybe the simplest to understand :beers:
And if we can auto-check `GITGUI_EDITMSG`, it will be the easiest to extend :wink: